File: applications/luci-app-statistics/luasrc/model/cbi/luci_statistics/nut.lua
-- Licensed to the public under the Apache License 2.0.

m = Map("collectd",
	translate("UPS Plugin Configuration"),
	translate("The NUT plugin reads information about Uninterruptible Power Supplies."))

s = m:section(NamedSection, "nut", "plugin" )

enable = s:option(Flag, "enable", translate("Enable this plugin"))
enable.default = 0

host = s:option(Value, "UPS", translate("UPS"), translate("UPS name in NUT ups@host format"))
host.placeholder = "myupsname"
host.datatype = "string"
host.rmempty = true

return m

File: applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/nut.js
'use strict';
'require baseclass';
'require form';

return baseclass.extend({
	title: _('UPS Plugin Configuration'),
	description: _('The NUT plugin reads information about Uninterruptible Power Supplies.'),

	addFormOptions(s) {
		s.option(form.Flag, 'enable', _('Enable this plugin'));

		s.option(form.Value, 'UPS', _('UPS'), _('UPS name in NUT ups@host format'));
	},

	configSummary(section) {
		const ups = L.toArray(section.UPS);

		if (ups.length)
			return N_(ups.length, 'Monitoring one UPS', 'Monitoring %d UPSes').format(ups.length);
	}
});
